File: java/Roman_to_Integer.java
public class Solution {
    public int romanToInt(String s) {
        
        ArrayList<Integer> l = new ArrayList<Integer>();
        
        for (int i = 0; i < s.length(); i++) {
            char item = s.charAt(i);
            if (item == 'I')
                l.add(1);
            if (item == 'V')
                l.add(5);
            if (item == 'X')
                l.add(10);
            if (item == 'L')
                l.add(50);
            if (item == 'C')
                l.add(100);
            if (item == 'D')
                l.add(500);
            if (item == 'M')
                l.add(1000);
        }
            
        int length = l.size();
        if(length == 1)
            return l.get(0);
        int Sum = 0;
        for (int j = 0; j < length; j++) {
            if(j < length - 1 && l.get(j) < l.get(j+1)){
                Sum -= l.get(j);
            }
            else
                Sum += l.get(j);
        }
        return Sum;
    }
}

File: python/Roman_to_Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        l = list()
        
        for item in s:
            if item == 'I':
                l.append(1)
            if item == 'V':
                l.append(5)
            if item == 'X':
                l.append(10)
            if item == 'L':
                l.append(50)
            if item == 'C':
                l.append(100)
            if item == 'D':
                l.append(500)
            if item == 'M':
                l.append(1000)
        length = len(l)
        if(length == 1):
            return l[0]
        Sum = 0
        for i in range(length):
            if(i < length - 1 and l[i] < l[i+1]):
                Sum -= l[i]
            else:
                Sum += l[i]
        return Sum
        
